src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProjectFactory.cs

```diff
@@ -803,44 +803,48 @@ public static PortableExecutableReference CreateReference_NoLock(string fullFile
         return reference;
     }
 
-#pragma warning disable VSTHRD100 // Avoid async void methods
-    private async void StartRefreshingMetadataReferencesForFile(object? sender, string fullFilePath)
-#pragma warning restore VSTHRD100 // Avoid async void methods
+    private void StartRefreshingMetadataReferencesForFile(object? sender, string fullFilePath)
     {
-        using var asyncToken = WorkspaceListener.BeginAsyncOperation(nameof(StartRefreshingMetadataReferencesForFile));
+        var asyncToken = WorkspaceListener.BeginAsyncOperation(nameof(StartRefreshingMetadataReferencesForFile));
+        var task = StartRefreshingMetadataReferencesForFileAsync(sender, fullFilePath);
+        task.CompletesAsyncOperation(asyncToken);
```

Comment:
❔ Do we want to also add a telemetry handler here?

Reply:
i'm ok with this as is for now.  unless there's a nice easy one we can hook up.